### PR TITLE
Added central configuration module

### LINKED ISSUE #4805: Pull configuration values to ouroboros-network
There are various default configuration values defined in different places.  I suggest we provide:
```
Ouroboros.Network.Diffusion.Configuration
```
module in  `ouroboros-network` which provides them.  Some of the defaults are defined in
* [cardano-node](https://github.com/IntersectMBO/cardano-node/blob/master/cardano-node/src/Cardano/Node/Configuration/POM.hs#L484-L526) and include:
  * `pncProtocolIdleTimeout`
  * `pncTimeWaitTimeout`
  * `pncAcceptedConnectionsLimit`
  * `pncTargetNumberOfRootPeers`
  * `pncTargetNumberOfKnownPeers`
  * `pncTargetNumberOfEstablishedPeers`
  * `pncTargetNumberOfActivePeers`
  * `pncTargetNumberOfKnownBigLedgerPeers`
  * `pncTargetNumberOfEstablishedBigLedgerPeers`
  * `pncTargetNumberOfActiveBigLedgerPeers`
  * `pncEnableP2P`
  * `pncPeerSharing`
  
from `ouroboros-network`:
  * `Ouroboros.Network.NodeToNode.defaultMiniProtocolParameters` which is used in [`miniProtocolParameters`](https://github.com/IntersectMBO/ouroboros-consensus/blob/a6e6707974d12fecd40650e6fb995d2058772d37/ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs#L667)
  * some values defined in [`Ouroboros.Network.Diffusion.Policies`](https://github.com/IntersectMBO/ouroboros-network/blob/f83f71ad674721605efa0f31cf2144cc6dd5dd4d/ouroboros-network/src/Ouroboros/Network/Diffusion/Policies.hs#L4) should be moved to the new module:
    * `deactivateTimeout`
    * `closeConnectionTimeout`
    * `peerMetricsConfiguration`

from `ouroboros-consensus`:
  * [`defaultBlockFetchConfiguration`](https://github.com/IntersectMBO/ouroboros-consensus/blob/a6e6707974d12fecd40650e6fb995d2058772d37/ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs#L672-L679)
  * [`stdChainSyncTimeout`](https://github.com/IntersectMBO/ouroboros-consensus/blob/a6e6707974d12fecd40650e6fb995d2058772d37/ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs#L724-L749)

### PR DESCRIPTION
**Pull request by crocodile-dentist**
# Description

resolves #4805 

# Checklist

- Branch
    - [ ] Updated changelog files.
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:69]: I don't think we should bring all this stuff from `cardano-node`.

Some of them are already defined in `ouroboros-network`, e.g.
* [connection manager][https://github.com/intersectmbo/ouroboros-network/blob/master/ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs#L446-L458] - they probably need to be defined in `ouroboros-network-framework`, then we just need to re-export them.
* We already have `PeerSelectionTargets` record, let's just define `defaultPeerSelectionTargets` in this module, and in the `cardano-node` we can use it in `mkDefaultNetworkConfig`.
* I found one more constant in [Handshake][https://github.com/intersectmbo/ouroboros-network/blob/f83f71ad674721605efa0f31cf2144cc6dd5dd4d/ouroboros-network-framework/src/Ouroboros/Network/Protocol/Handshake.hs#L190-L191].

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:69]: Note that the advantage of re-using `PeerSelectionTarget`, is that whenever we modify it we will be forced to add a default value. @bolt12 changes to bootstrap are of this kind.

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:22]: @bolt12 are these values used in `cardano-node` or are these only internals.

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:4]: It's better to explicitly export.  It would also be good to add haddock sections, some of them fall into different groups: `OutboundGovernor`, `InboundGovernor`, `Protocol params`.

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:47]: hmm, what `Qty` stands for?

**Comment by coot** [comment]
Once this is merged, we should integrate the changes to `ouroboros-consensus` and `cardano-node`.  I'd prefer if we do that than leave it to the next `release-engineer`.

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (bolt12) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:22]: They are used in `ouroboros-consensus`

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (bolt12) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:47]: Does this comes from the `targetNumberOfBootstrapPeers` if so, this is no longer needed as a configuration.

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (crocodile-dentist) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:4]: In general you're right, the reason is when I was writing it I came to the conclusion that this sort of configuration module consisting of defaults and constants is a special case where I imagine we probably will end up exporting everything anyhow. But I will change it to explicit export list.

File: ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs
Comment (crocodile-dentist) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:47]: It's an established abbreviation for quantity, the type was getting a little long with Number and I thought this was a good synonym. Should I change to Number or Count?

Comment (bolt12) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:47]: what's this config about?

Comment (crocodile-dentist) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs:47]: Based on my discussion with @coot we want to keep this constant around for some time even after transitioning to genesis but it will go away at some point.

File: ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Diffusion/Node/NodeKernel.hs
Comment (coot) [inline ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Diffusion/Node/NodeKernel.hs:369]:
```suggestion
    let (bpSeed, psSeed) = split bgaSeed
```
Then `bpSeed` should be used below when passing to `loop` instead of `bgaSeed`.  Splitting an RNG is deterministic, so in the current implementation we might get correlations between two `rng`s that should be independent if it is split in the loop again.

**Review by coot — CHANGES_REQUESTED** [review]

File: ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs
Comment (crocodile-dentist) [reply ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs:666]: These are @bolt12 changes from #4795

File: ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Diffusion/Node/NodeKernel.hs
Comment (crocodile-dentist) [reply ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Diffusion/Node/NodeKernel.hs:369]: same as above

**Review by coot — APPROVED** [review]
LGTM; #4795 was merged, once conflicts are resolved this can be merged.

**Pull request merged**
